VEX/pub/libvex_basictypes.h

```diff
@@ -90,8 +90,11 @@ typedef
    V256;
 
 /* Floating point. */
-typedef  float   Float;    /* IEEE754 single-precision (32-bit) value */
-typedef  double  Double;   /* IEEE754 double-precision (64-bit) value */
+#ifdef __riscv_zfh
+typedef  _Float16 Float16;  /* IEEE754 half-precision   (16-bit) value */
```

Comment:
_Float16 isn't RISCV specific. I'm not sure which version of C added it. From what I see there is no straightforward feature test for this. See https://stackoverflow.com/questions/69977196/how-to-correctly-determine-at-compile-time-that-float16-is-supported

It might be worth adding a configure time check for this.